File: Data Structures & Algorithms/same-binary-tree/submission-6.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

func isSameTree(p *TreeNode, q *TreeNode) bool {
	//base conditions 
	if p == nil && q == nil{
		return true
	}
	if (p == nil && q != nil) || (p != nil && q == nil){
		return false
	}
	if p.Val != q.Val {
		return false
	}

	l := isSameTree(p.Left, q.Left)
	r := isSameTree(p.Right, q.Right)

	// operations



	return l && r
    
}

File: Data Structures & Algorithms/binary-tree-right-side-view/submission-6.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def rightSideView(self, root: Optional[TreeNode]) -> List[int]:
        self.ans = []
        def dfs(root, level):
            if root is None:
                return
            if len(self.ans) == level:
                self.ans.append([])
            self.ans[level] = root.val
            dfs(root.left, level+1)
            dfs(root.right, level+1)
        dfs(root, 0)
        return self.ans
            

            
        